Tests/test_PDB.py

```diff
@@ -995,6 +995,78 @@ def test_get_atoms(self):
         self.assertEqual(len(atoms), 756)
 
 
+class ChangingIdTests(unittest.TestCase):
+
+    def setUp(self):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", PDBConstructionWarning)
+            self.struc = PDBParser(PERMISSIVE=True).get_structure(
+                                                  'X', "PDB/a_structure.pdb")
+
+    def test_change_model_id(self):
+        """Change the id of a model"""
+        for model in self.struc:
+            break  # Get first model in structure
+        model.id = 2
+        self.assertEqual(model.id, 2)
+        self.assertIn(2, self.struc)
+        self.assertNotIn(0, self.struc)
+
+    def test_change_model_id_raises(self):
+        """Cannot change id to a value already in use by another child"""
+        for model in self.struc:
+            break  # Get first model in structure
+        with self.assertRaises(ValueError):
+            model.id = 1
+        # Make sure nothing was changed
+        self.assertEqual(model.id, 0)
+        self.assertIn(0, self.struc)
+        self.assertIn(1, self.struc)
+
+    def test_change_chain_id(self):
+        """Change the id of a model"""
+        for chain in self.struc.get_chains():
+            break  # Get first chain in structure
+        chain.id = "R"
+        self.assertEqual(chain.id, "R")
+        for model in self.struc:
+            break  # Get first model
+        self.assertIn("R", model)
+
+    def test_change_residue_id(self):
+        """Change the id of a residue"""
+        for chain in self.struc.get_chains():
+            break  # Get first chain in structure
+        res = chain[('H_PCA', 1, ' ')]
+        res.id = (' ', 1, ' ')
+
+        self.assertEqual(res.id, (' ', 1, ' '))
+        self.assertIn((' ', 1, ' '), chain)
+        self.assertNotIn(('H_PCA', 1, ' '), chain)
+        self.assertEqual(chain[(' ', 1, ' ')], res)
+
+    def test_full_id_is_updated(self):
+        """
+        Invalidate cached full_ids if an id is changed.
+        """
+        for atom in self.struc.get_atoms():
+            break  # First atom
+        original_id = atom.get_full_id()  # ('X', 0, 'A', ('H_PCA', 1, ' '), ('N', ' '))
+        if original_id != ('X', 0, 'A', ('H_PCA', 1, ' '), ('N', ' ')):
```

Comment:
I did not use `assertEqual` on purpose, because I am asserting something about the file I load for the test, not about the functionality I am testing, so I thought letting the test result in an Error not a Failure is clearer. (There is currently no test addressing the caching of full_ids)

The way I wrote the tests (depending on a loaded pdb file instead of creating an instance of Entity and mocking all external dependencies) is in accordance with the other tests in test_PDB and somewhat more practical/ easier, but it brings clear disadvantages, such as depending on the content of the file.